src/myrtlespeech/data/sampler.py

```diff
@@ -23,10 +56,49 @@ def _batch_indices(self, indices, batch_size, drop_last):
         return batches
 
     def __iter__(self):
-        if self.shuffle:
-            random.shuffle(self.batch_indices)
-        for b in self.batch_indices:
-            yield b
+        indices = list(range(len(self.batch_indices)))
+        if self.shuffle and self._n_iterators not in self._sequential:
+            random.shuffle(indices)
+        self._n_iterators += 1
+        for index in indices:
+            yield self.batch_indices[index]
 
     def __len__(self):
         return len(self.batch_indices)
+
+
+class SortaGrad(SequentialRandomSampler):
+    """An iterable over batch indices according to the SortaGrad strategy.
+
+    The SortaGrad curriculum learning strategy iterates over batches from the
+    batched dataset sequentially for the first pass and then randomly for all
+    other passes. See Deep Speech 2 paper for more information on this:
+    https://arxiv.org/abs/1512.02595
```

Comment:
Restructured Text has syntax for creating a hyperlink, see [here](http://openalea.gforge.inria.fr/doc/openalea/doc/_build/html/source/sphinx/rest_syntax.html#id6):

```
`Deep Speech 2 <https://arxiv.org/abs/1512.02595>`_ 
```